src/modes/ModeEstimation.jl

```diff
@@ -0,0 +1,235 @@
+
+    # Retrieve the estimated values.
+    vals = binv(M.minimizer)
+
+    # Get the VarInfo at the MLE/MAP point, and run the model to ensure 
+    # correct dimensionality.
+    vi[spl] = vals
+    model(vi) # XXX: Is this a necessary step?
+
+    # Make one transition to get the parameter names.
+    ts = [Turing.Inference.Transition(DynamicPPL.tonamedtuple(vi), DynamicPPL.getlogp(vi))]
+    varnames, _ = Turing.Inference._params_to_array(ts)
+
+    # Store the parameters and their names in an array.
+    vmat = NamedArrays.NamedArray(vals, varnames)
+
+    # Try to generate the information matrix.
+    try
+        # Calculate Hessian and information matrix.
+        info = ForwardDiff.hessian(hess_target, vals)
+        info = inv(info)
+        mat = NamedArrays.NamedArray(info, (varnames, varnames))
+
+        # Create the standard errors.
+        ses = sqrt.(diag(info))
+
+        # Calculate t-stats.
+        tstat = vals ./ ses
+
+        # Make a summary table.
+        stable = NamedArrays.NamedArray(
+            [vals ses tstat], 
+            (varnames, ["parameter", "std_err", "tstat"]))
+
+        # Return a wrapped-up table.
+        return ModeResult(vmat, mat, M, stable, M.minimum)
+    catch err
+        @warn "Could not compute Hessian matrix" err
+        stable = NamedArrays.NamedArray([vals repeat([missing], length(vals)) repeat([missing], length(vals))], (varnames, ["parameter", "std_err", "tstat"]))
+        return ModeResult(vmat, missing, M, stable, M.minimum)
+    end
+end
+
+"""
+    MLE(model::DynamicPPL.Model; kwargs...)
+
+Returns a maximum likelihood estimate of the given `model`.
+
+Arguments: 
+
+- `model` is a `DynamicPPL.Model`.
+
+Keyword arguments:
+
+- `optim_options` is a `Optim.Options` struct that allows you to change the number
+  of iterations run in an MLE estimate.
+
+Usage:
+
+```julia
+using Turing
+
+@model function f()
+    m ~ Normal(0, 1)
+    1.5 ~ Normal(m, 1)
+    2.0 ~ Normal(m, 1)
+end
+
+model = f()
+mle_estimate = MLE(model)
+
+# Manually setting the optimizers settings.
+mle_estimate = MLE(model, optim_options=Optim.Options(iterations=500))
+```
+"""
+function MLE(model::DynamicPPL.Model; kwargs...)
+    lpf = make_logjoint(model, DynamicPPL.LikelihoodContext())
+    return mode_estimation(model, lpf; kwargs...)
+end
+
+"""
+    MAP(model::DynamicPPL.Model; kwargs...)
+
+Returns the maximum a posteriori estimate of the given `model`.
+
+Arguments: 
+
+- `model` is a `DynamicPPL.Model`.
+
+Keyword arguments:
+
+- `optim_options` is a `Optim.Options` struct that allows you to change the number
+  of iterations run in an MLE estimate.
+
+Usage:
+
+```julia
+using Turing
+
+@model function f()
+    m ~ Normal(0, 1)
+    1.5 ~ Normal(m, 1)
+    2.0 ~ Normal(m, 1)
+end
+
+model = f()
+mle_estimate = MAP(model)
+
+# Manually setting the optimizers settings.
+mle_estimate = MAP(model, optim_options=Optim.Options(iterations=500))
+```
+"""
+function MAP(model::DynamicPPL.Model; kwargs...)
```

Comment:
This should be a function as well but unfortunately it can't be named `map`... Maybe use something more explicit?